pkg/storage/module/upload.go

```diff
@@ -14,16 +15,42 @@ import (
 const numFiles = 3
 
 // Uploader takes a stream and saves it to the blob store under a given path
-type Uploader func(ctx context.Context, path, contentType string, stream io.Reader) error
+type Uploader func(ctx context.Context, path, contentType string, stream Stream) error
+
+// Stream is the object that is passed along to Uploaders
+type Stream struct {
+	Stream io.Reader
+	Size   int64
+}
+
+// NewStreamFromBytes returns a new module.Stream from a slice of bytes to Savers
+func NewStreamFromBytes(b []byte) Stream {
+	return Stream{
+		Stream: bytes.NewReader(b),
+		Size:   int64(len(b)),
+	}
+}
+
+// NewStreamFromReaderWithSize returns a new module.Stream from an io.Reader and keeps the len -1 for minio storage > 600 MB
+func NewStreamFromReaderWithSize(r io.Reader, s int64) Stream {
+	const MinioLimit = 600000000
+	if s > MinioLimit {
```

Comment:
can you unit test this?

Reply:
also how will minio behave when setting to 600 and having a module which is >600?